syntax = "proto3";

import "google/protobuf/any.proto";

service KVStore {
    rpc Set(SetRequest) returns (SetResponse) {}
    rpc Get(GetRequest) returns (GetResponse) {}
    rpc Remove(RemoveRequest) returns (RemoveResponse) {}
}

message SetRequest {
    string key = 1;
    google.protobuf.Any value = 2;
}

message SetResponse {
    bool success = 1;
}

message GetRequest {
    string key = 1;
}

message GetResponse {
    google.protobuf.Any value = 1;
}

message RemoveRequest {
    string key = 1;
}

message RemoveResponse {
    bool success = 1;
}